File: ios/LinkU/Models/Task.swift
import Foundation

struct Task: Codable, Identifiable {
    let id: Int
    let title: String
    let description: String
    let category: String
    let city: String
    let price: Double?
    let status: TaskStatus
    let images: [String]?
    let createdAt: String
    let author: User?
    
    enum CodingKeys: String, CodingKey {
        case id, title, description, category, city, price, status, images
        case createdAt = "created_at"
        case author
    }
}

enum TaskStatus: String, Codable {
    case open = "open"
    case inProgress = "in_progress"
    case completed = "completed"
    case cancelled = "cancelled"
    
    var displayText: String {
        switch self {
        case .open: return "开放中"
        case .inProgress: return "进行中"
        case .completed: return "已完成"
        case .cancelled: return "已取消"
        }
    }
}

struct TaskListResponse: Codable {
    let tasks: [Task]
    let total: Int
    let page: Int
    let pageSize: Int
    
    enum CodingKeys: String, CodingKey {
        case tasks
        case total
        case page
        case pageSize = "page_size"
    }
}

File: android/app/src/main/java/com/link2ur/app/data/models/Task.kt
package com.linku.app.data.models

import com.google.gson.annotations.SerializedName

data class Task(
    val id: Int,
    val title: String,
    val description: String,
    @SerializedName("task_type")
    val taskType: String,
    val location: String,
    val reward: Double,
    val status: TaskStatus,
    @SerializedName("created_at")
    val createdAt: String,
    @SerializedName("updated_at")
    val updatedAt: String?,
    @SerializedName("poster_id")
    val posterId: String,
    @SerializedName("taker_id")
    val takerId: String?,
    val images: List<String>?,
    val deadline: String?,
    @SerializedName("is_flexible")
    val isFlexible: Int?
) {
    // 兼容属性
    val category: String get() = taskType
    val city: String get() = location
    val price: Double? get() = reward
}

enum class TaskStatus {
    @SerializedName("open")
    OPEN,
    @SerializedName("in_progress")
    IN_PROGRESS,
    @SerializedName("completed")
    COMPLETED,
    @SerializedName("cancelled")
    CANCELLED
}

data class TaskListResponse(
    val tasks: List<Task>,
    val total: Int,
    val page: Int,
    @SerializedName("page_size")
    val pageSize: Int
)

data class CreateTaskRequest(
    val title: String,
    val description: String,
    @SerializedName("task_type")
    val taskType: String,
    val location: String,
    val reward: Double,
    val images: List<String>?,
    val deadline: String?,
    @SerializedName("is_flexible")
    val isFlexible: Int,
    @SerializedName("is_public")
    val isPublic: Int
)

